### PR TITLE
bugfix: improve 'Time Table'

### LINKED ISSUE #6948: [time series table] On viz type change default to adding a Time Series column entry
When creating a time series table, a metric is already populated in the Metrics field, but the visualization appears to just keep on loading, when in reality, a Time Series column needs to be created. This gives the user the appearance that he/she should wait until something loads.

Expected behavior should be:
(1) Do not show 'loading' Superset icon. Add some sort of indication that a time series column needs to be added
(2) Default to the metric in the metric field as the time series column since that is already populated

Make sure these boxes are checked before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

### Superset version
v.031

### Steps to reproduce
(1) Sources->Tables-> Choose a datasource
(2) In Explore View, switch viz type to Time Series Table
(3) Note Superset logo loading icon keeps going

### PR DESCRIPTION
**Pull request by mistercrunch**
closes https://github.com/apache/incubator-superset/issues/6948

@vylc was able to reproduce and also hit unrelated issues around `Time Table`. Started work here but reached out for help on Slack.

screen shot 2019-02-27 at 11 29 44 pm

### CODE REVIEW

File: superset/assets/src/explore/components/SaveModal.jsx [hunk 1]
```diff
@@ -124,7 +124,8 @@ class SaveModal extends React.Component {
     }
     sliceParams.goto_dash = gotodash;
 
-    this.props.actions.saveSlice(this.props.form_data, sliceParams).then(({ data }) => {
+    this.props.actions.saveSlice(this.props.form_data, sliceParams).then((payload) => {
```
Comment (xtinec) [inline superset/assets/src/explore/components/SaveModal.jsx]: hmm, this should have destructed properly without any change. Did it not?

**Comment by michellethomas** [comment]
@mistercrunch have you dug into this anymore? We are running into the error `Converting circular structure to JSON` in 0.31 with adding anything to time table.

When I remove [this line](https://github.com/apache/incubator-superset/pull/6523/files#diff-11e98750ad920e62b68d5aadc8128f16R86) I don't get the error. It looks like all props are getting added to each object in `value`, which creates a circular reference where each object in `value` contains `value` (the list of time series column params). Do you remember why you were adding props to the control here?

**Comment by mistercrunch** [comment]
@michellethomas this is ready for review!

File: superset/assets/src/SqlLab/components/ExploreResultsButton.jsx [hunk 2]
```diff
@@ -141,7 +141,9 @@ class ExploreResultsButton extends React.PureComponent {
         // open new window for data visualization
         exportChart(formData);
       })
-      .catch(() => {
+      .catch((e) => {
+        /* eslint no-console: 0 */
+        console.error(e);
```
Comment (michellethomas) [inline superset/assets/src/SqlLab/components/ExploreResultsButton.jsx]: Did you intend to include these console messages?

**Review by michellethomas — APPROVED** [review]
thanks for fixing this, one question about the console logs but lgtm otherwise

File: superset/assets/src/SqlLab/components/ExploreResultsButton.jsx (hunk 2 above)
Comment (mistercrunch) [reply superset/assets/src/SqlLab/components/ExploreResultsButton.jsx]: I did, but after reading a bit about best practices, I removed it...

**Pull request merged**